### PR TITLE
Mount azure files

### PR DESCRIPTION
**Pull request by manuelesimi**
This ansible playbook mounts an Azure Files endpoint on selected nodes. Obviously it works only  when the cluster is created on Azure.

It requires the user to tag the nodes/groups with `azure-files` and set the following global variables in the `setup` section of the cluster:

```
[setup/mycluster ]
provider=ansible
frontend_groups=..
compute_groups=...,azure-files

# information to mount an Azure Files endpoint
# the storage account name with the files
global_var_af_storage_account_name=
# the name used to share the Azure files
global_var_af_share_name=
# the account key fot the storage
global_var_af_storage_account_key=
# A folder for a mount point. It can be created anywhere on the file system, but it's common convention to create this under the /mnt folder.
global_var_af_mount_point=

```
Reference documentation: https://docs.microsoft.com/en-us/azure/storage/files/storage-how-to-use-files-linux

### CODE REVIEW

File: elasticluster/share/playbooks/roles/azure-files/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+- name: Create Azure files destination directory
+  file:
+    path: "{{ mount_point }}"
+    state: directory
+    mode: "u=rw,g=rw,o=rw"
+    recurse: yes
+  become: true
+
+- name: Concatenate mount options
+  set_fact:
```
Comment (riccardomurri) [inline elasticluster/share/playbooks/roles/azure-files/tasks/main.yml]: Why do you need to set this as a fact/variable? Can't you just paste the definition in here onto line 27?

File: elasticluster/share/playbooks/roles/azure-files/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    mount_opts_cat: "{{'vers=3.0,username=' + account +',password=' + password +',dir_mode=0777,file_mode=0777,serverino'}}"
```
Comment (riccardomurri) [inline elasticluster/share/playbooks/roles/azure-files/tasks/main.yml]: Better:
```suggestion
    mount_opts_cat: "vers=3.0,username={{account}},password={{password}},dir_mode=0777,file_mode=0777,serverino"
```

File: elasticluster/share/playbooks/roles/azure-files/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+- name: Mount Azure Files endpoint
+  mount:
+    path: "{{ mount_point }}"
+    src: "//{{ account }}.file.core.windows.net/{{ share_name }}"
+    fstype: "cifs"
+    opts: "{{ mount_opts_cat }}"
```
Comment (riccardomurri) [inline elasticluster/share/playbooks/roles/azure-files/tasks/main.yml]: Same as for `mount_point`: should this be renamed `af_mount_opts` or similar? (But see comment to line 19)

File: elasticluster/share/playbooks/roles/azure-files/vars/main.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+account: "{{ af_storage_account_name }}"
```
Comment (riccardomurri) [inline elasticluster/share/playbooks/roles/azure-files/vars/main.yml]: The whole `var/main.yml` file looks superfluous to me -- just use the `af_*` variables in the tasks.

File: elasticluster/share/playbooks/roles/azure-files/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  become: true
+
+- name: Concatenate mount options
+  set_fact:
+    mount_opts_cat: "{{'vers=3.0,username=' + account +',password=' + password +',dir_mode=0777,file_mode=0777,serverino'}}"
+
+- name: Mount Azure Files endpoint
+  mount:
+    path: "{{ mount_point }}"
+    src: "//{{ account }}.file.core.windows.net/{{ share_name }}"
```
Comment (riccardomurri) [inline elasticluster/share/playbooks/roles/azure-files/tasks/main.yml]: See comment regarding file `vars/main.yml` -- just use the `af_*` variables here.

**Comment by manuelesimi** [comment]
Thanks for reviewing the request. The comments make sense to me. I'll address them and push the changes to my branch. Thanks!

**Comment by manuelesimi** [comment]
Please, see ef86673.

**Comment by riccardomurri** [comment]
I've added a few things (mainly docs and conditionals) here: https://github.com/gc3-uzh-ch/elasticluster/compare/master...riccardomurri:pr/%23639?expand=1

Can you give my branch `pr/#639` a try?  I'll merge if you report it still works.

**Comment by manuelesimi** [comment]
I tried your branch. Changing the state from `mounted` to `present` does not work. 
`present` just creates the entry in the fstab and the mount dir but it doesn't not really mount the azure files location. As a consequence, the files are not available when the playbook returns.

See:
https://docs.ansible.com/ansible/latest/modules/mount_module.html

**Comment by riccardomurri** [comment]
Thanks, I recalled there was a distinction but didn't bother to check the docs so I picked the wrong one.

**Comment by riccardomurri** [comment]
I've now squash-merged your commits, together with mine, into "master".

ElastiCluster 1.3.dev15 is the first tag to enjoy the new "Azure files" feature.

Many many thanks for contributing yet another useful feature to ElastiCluster!

**Comment by manuelesimi** [comment]
Thanks!

**Pull request closed without merging**